### PR TITLE
[stripe] Upgrade stripe to 0.23.1

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

https://linear.app/expo/issue/ENG-7314/update-stripestripe-react-native

# How

https://www.notion.so/expo/Updating-stripe-react-native-for-a-new-SDK-release-646ee16729c844e98b8c700d0c04dd1b

# Test Plan
Running stripe example app in Expo Go unversioned together with the stripe server following this AWESOME guide from @Kudo https://github.com/expo/expo/pull/19432
image
image

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by aleqsio** [comment]
[ISSUE]
This required manually modifying headers in Pods, working on this with @tsapeta

File: ios/vendored/unversioned/@stripe/stripe-react-native/ios/Tests/AddressSheetUtilsTests.swift [hunk 1]
```diff
@@ -0,0 +1,273 @@
+//
```
Comment (Kudo) [inline ios/vendored/unversioned/@stripe/stripe-react-native/ios/Tests/AddressSheetUtilsTests.swift:1]: we could exclude these test files in expoGoConfig.ts

File: ios/Podfile [hunk 2]
```diff
@@ -94,6 +94,13 @@ abstract_target 'Expo Go' do
         end
       end
 
+      if pod_name == 'StripePaymentsUI'
+        target_installation_result.native_target.build_configurations.each do |config|
```
Comment (Kudo) [inline ios/Podfile:98]: maybe add a comment here to describe why we need this workaround.

File: ios/vendored/unversioned/@stripe/stripe-react-native/ios/Tests/AddressSheetUtilsTests.swift (hunk 1 above)
Comment (aleqsio) [reply ios/vendored/unversioned/@stripe/stripe-react-native/ios/Tests/AddressSheetUtilsTests.swift:1]: Exclude does not work, seems that `copyVendoredFilesAsync` does not take this `excludeFiles` into account. I just removed manually for now

Comment (Kudo) [reply ios/vendored/unversioned/@stripe/stripe-react-native/ios/Tests/AddressSheetUtilsTests.swift]: oh yep, the `excludeFiles` on ios was just added last month and only available for *.podspec glob. thanks for checking this.

**Review by Kudo — APPROVED** [review]

**Comment by Kudo** [comment]
thanks for helping this tough module ❤️ 👏

**Pull request merged**